src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -438,6 +438,11 @@ private TypeWithAnnotations ComputeType(Binder binder, SyntaxNode syntax, Bindin
                 diagnostics.Add((this.IsIndexer ? ErrorCode.ERR_BadVisIndexerReturn : ErrorCode.ERR_BadVisPropertyType), Location, this, type.Type);
             }
 
+            if (type.Type.IsFileTypeOrUsesFileTypes() && !ContainingType.IsFileTypeOrUsesFileTypes())
```

Comment:
We probably need to care for `GetExplicitInterfaceSpecifier()` like we do in case above. This suggests to me that folding logic into `IsNoMoreVisibleThan` might be a good idea.

Reply:
Discussion in https://github.com/dotnet/roslyn/pull/60977#discussion_r880960599